e2e/internal/contrasttest/contrasttest.go

```diff
@@ -374,14 +398,29 @@ func (ct *ContrastTest) FactorPlatformTimeout(timeout time.Duration) time.Durati
 }
 
 // MakeNamespace creates a namespace string using a given *testing.T.
-func MakeNamespace(t *testing.T) string {
-	buf := make([]byte, 4)
+func MakeNamespace(t *testing.T, namespaceSuffix string) string {
+	var namespaceParts []string
+
+	// First part(s) are consist of all valid characters in the lower case test name.
 	re := regexp.MustCompile("[a-z0-9-]+")
+	namespaceParts = append(namespaceParts, re.FindAllString(strings.ToLower(t.Name()), -1)...)
+
+	// Append some randomness
+	buf := make([]byte, 4)
 	n, err := rand.Reader.Read(buf)
 	require.NoError(t, err)
 	require.Equal(t, 4, n)
 
-	return strings.Join(append(re.FindAllString(strings.ToLower(t.Name()), -1), hex.EncodeToString(buf)), "-")
+	namespaceParts = append(namespaceParts, fmt.Sprintf("%x", buf))
+
+	// Below we join all parts with a dash already, the user usually provides the
+	// suffix with a dash, so we remove any leading dash from the suffix.
+	namespaceSuffix = strings.TrimPrefix(namespaceSuffix, "-")
+
+	// Append the suffix
+	namespaceParts = append(namespaceParts, namespaceSuffix)
```

Comment:
If we don't treat the suffix as literal you can't leave it empty, because a trailing dash violates https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names. Can we just append it literally?

Reply:
done